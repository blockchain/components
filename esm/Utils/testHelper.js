import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import theme from '../Tools/theme';

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(React.createElement(ThemeProvider, {
    theme: theme
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

export function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [shallow].concat(Array.prototype.slice.call(arguments)));
}
export function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [mount].concat(Array.prototype.slice.call(arguments)));
}