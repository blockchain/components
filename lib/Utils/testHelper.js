"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.shallowWithTheme = shallowWithTheme;
exports.mountWithTheme = mountWithTheme;

var _react = _interopRequireDefault(require("react"));

var _enzyme = require("enzyme");

var _styledComponents = require("styled-components");

var _theme = _interopRequireDefault(require("../Tools/theme"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(_react.default.createElement(_styledComponents.ThemeProvider, {
    theme: _theme.default
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [_enzyme.shallow].concat(Array.prototype.slice.call(arguments)));
}

function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [_enzyme.mount].concat(Array.prototype.slice.call(arguments)));
}