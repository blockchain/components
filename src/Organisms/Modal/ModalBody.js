// @flow strict
import PropTypes from 'prop-types'
import * as React from 'react'
import styled from 'styled-components'
import { screenSize } from '../../Tools/interpolation'

const Wrapper = styled.div`
  width: 100%;
  height: 100%;
  overflow: auto;
  padding: 1.5rem;
  box-sizing: border-box;

  @media (min-width: ${screenSize('sm')}) {
    height: auto;
    overflow-y: none;
  }
`

export type PropsType = {
  +children?: React.Node,
}

const ModalBody = ({ children, ...rest }: PropsType) => (
  <Wrapper {...rest}>{children}</Wrapper>
)

ModalBody.propTypes = {
  children: PropTypes.node,
}

export default ModalBody
