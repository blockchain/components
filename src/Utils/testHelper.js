import { mount } from 'enzyme'
import { ThemeConsumer } from 'styled-components'
import theme from '../Tools/theme'

export const mountWithTheme = (children) => {
  ThemeConsumer._currentValue = theme
  return mount(children)
}
