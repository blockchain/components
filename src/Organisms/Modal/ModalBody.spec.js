import React from 'react'
import { mountWithTheme } from '../../Utils/testHelper'

import ModalBody from './ModalBody'

describe('ModalBody', () => {
  it('should render correctly', () => {
    const tree = mountWithTheme(<ModalBody />)

    expect(tree.find(ModalBody)).toMatchSnapshot()
  })
})
