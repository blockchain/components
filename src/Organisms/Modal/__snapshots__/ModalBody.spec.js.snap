// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalBody should render correctly 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  overflow: auto;
  padding: 1.5rem;
  box-sizing: border-box;
}

@media (min-width:48rem) {
  .c0 {
    height: auto;
    overflow-y: none;
  }
}

<ModalBody>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  width: 100%;
  height: 100%;
  overflow: auto;
  padding: 1.5rem;
  box-sizing: border-box;

  @media (min-width: ",
              [Function],
              ") {
    height: auto;
    overflow-y: none;
  }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="c0"
      />
    </StyledComponent>
  </styled.div>
</ModalBody>
`;
